### PR TITLE
Prioritization changes notifies other users 

### PR DESCRIPTION
**Pull request by taksan**

### CODE REVIEW

File: src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java [hunk 1]
```diff
@@ -84,27 +80,16 @@ public void add(WebhookEvent event, String issueKey) {
...
     public void processItems() {
         List<Item> toRemove = new ArrayList<Item>();
-        List<IssueUpdate> updatedIssues = new ArrayList<>();
 
+        issueBufferService.startBatchUpdate();
         for (Item item : getItens()) 
             try {
                 if (item.event.isTypeVersion()) {
                     cacheManager.getCache(PROJECTS).clear();
                     if (item.event.equals(VERSION_UPDATED))
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java]: CRITICAL Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS134)

File: src/main/java/objective/taskboard/database/IssuePriorityService.java [hunk 2]
```diff
@@ -101,10 +110,14 @@ public synchronized void reorder(String[] issueKeys) {
         
         for (String pkey : issueKeys) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/database/IssuePriorityService.java]: MINOR Combine this loop with the one that starts on line 100. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS3047)

File: src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java [hunk 3]
```diff
@@ -127,6 +135,37 @@ public synchronized Issue updateIssueBuffer(WebhookEvent event, final String key
...
+
+        return updated;
+    }
+
+    public synchronized void startBatchUpdate() {
+        issuesUpdatedByEvent.clear();
+    }
+
+    public synchronized void finishBatchUpdate() {
+        if (issuesUpdatedByEvent.size() > 0)
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java]: MINOR Use isEmpty() to check whether the collection is empty or not. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1155)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L53](https://github.com/objective-solutions/taskboard/blob/2134b59ebeffe11f6da501de80420f2a9870f7cb/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L53): Cast one of the operands of this multiplication operation to a "long". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2184)
1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L57](https://github.com/objective-solutions/taskboard/blob/2134b59ebeffe11f6da501de80420f2a9870f7cb/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L57): Make this a "static" inner class. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2694)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L53](https://github.com/objective-solutions/taskboard/blob/aabea1c88b9c01b5a113e6e7c0126d701860acf8/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L53): Cast one of the operands of this multiplication operation to a "long". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2184)
1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L57](https://github.com/objective-solutions/taskboard/blob/aabea1c88b9c01b5a113e6e7c0126d701860acf8/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L57): Make this a "static" inner class. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2694)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L53](https://github.com/objective-solutions/taskboard/blob/aabea1c88b9c01b5a113e6e7c0126d701860acf8/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L53): Cast one of the operands of this multiplication operation to a "long". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2184)
1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L57](https://github.com/objective-solutions/taskboard/blob/aabea1c88b9c01b5a113e6e7c0126d701860acf8/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L57): Make this a "static" inner class. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2694)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/test/java/objective/taskboard/it/ReprioritizationIT.java [hunk 4]
```diff
@@ -44,9 +48,62 @@ public void whenIssueIsDragged_AfterReloadItShouldKeepOrder() {
...
+                "TASKB-682",
+                "TASKB-683",
+                "TASKB-684",
+                "TASKB-686"
+                );
+    }
+
+    @Test
+    public void whenIssueIsPriorityOrderIsChanged_ShouldShowNotificationInAnotherBrowserAndUpdateTheOrder() {
+        webDriver.findElement(By.cssSelector("body")).sendKeys(Keys.CONTROL +"t");
```
Comment (lohandus) [inline src/test/java/objective/taskboard/it/ReprioritizationIT.java]: Minor: Essa infra de abrir uma nova aba e alterar o driver pra ela poderia estar no AbstractUIIntegrationTest. Nem tanto por causa do reuso, mas mais para tirar essa complexabilidade do código do teste.

File: src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java (hunk 3 above)
Comment (taksan) [inline src/main/java/objective/taskboard/issueBuffer/IssueBufferService.java]: Não foi sem querer. A intenção era reduzir ao mínimo o código condicional mesmo. Afinal, o clear é inócuo se for vazio e faz o que deve ser não for

File: src/main/java/objective/taskboard/database/IssuePriorityService.java [hunk 5]
```diff
@@ -74,18 +75,26 @@ public Long determinePriority(Issue e) {
         return priorityOrder.getPriority();
     }
 
-    public synchronized void reorder(String[] issueKeys) {
+    public Date priorityUpdateDate(Issue e) {
```
Comment (taksan) [inline src/main/java/objective/taskboard/database/IssuePriorityService.java]: Legal, acho que é melhor mesmo.

File: src/test/java/objective/taskboard/it/ReprioritizationIT.java (hunk 4 above)
Comment (taksan) [inline src/test/java/objective/taskboard/it/ReprioritizationIT.java]: Ok, inicialmente pensei que não seria reutilizado. Mas acho que vai ficar mais legível mesmo.

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L53](https://github.com/objective-solutions/taskboard/blob/0536fea302e84f4168927bfdd738ec4cfd9fd952/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L53): Cast one of the operands of this multiplication operation to a "long". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2184)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [IssueEventProcessScheduler.java#L53](https://github.com/objective-solutions/taskboard/blob/c542bea10780d3089fb6d2b8a7c6594c2239a213/src/main/java/objective/taskboard/task/IssueEventProcessScheduler.java#L53): Cast one of the operands of this multiplication operation to a "long". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2184)

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**